test_runner/regress/test_replication_start.py

```diff
@@ -0,0 +1,26 @@
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import NeonEnv, wait_replica_caughtup
+
+
+def test_replication_start(neon_simple_env: NeonEnv):
+    env = neon_simple_env
+
+    with env.endpoints.create_start(branch_name="main", endpoint_id="primary") as primary:
+        with primary.connect() as p_con:
+            with p_con.cursor() as p_cur:
+                p_cur.execute("begin")
+                p_cur.execute("create table t(pk integer primary key, payload integer)")
+                p_cur.execute("insert into t values (generate_series(1,100000), 0)")
+                p_cur.execute("select txid_current()")
+                xid = p_cur.fetchall()[0][0]
+                log.info(f"Master transaction {xid}")
+                with env.endpoints.new_replica_start(
+                    origin=primary, endpoint_id="secondary"
+                ) as secondary:
+                    wait_replica_caughtup(primary, secondary)
+                    with secondary.connect() as s_con:
+                        with s_con.cursor() as s_cur:
+                            s_cur.execute("select * from pg_class")
+                            p_cur.execute("commit")
+                            wait_replica_caughtup(primary, secondary)
+                            s_cur.execute("select * from t where pk = 1")
```

Comment:
You're not checking the results of these queries. I presume this test will pass with or without the code changes, which is not good. Please make sure the test fails without the fix.

Reply:
Without this fix table 't' is not failed at replica and test failed with error.

Reply:
But added check in any case...

Reply:
Please verify that this test fails without the code changes. I'm still not 100% convinced.

By the time you run the `select * from t where pk = 1` query, the standby has received the COMMIT record, and sees the transaction as visible, with or without this bug fix. The problem scenario we're fixing is the other way round. If you start a transaction and acquire a snapshot in the replica before the commit, and then use the snapshot after the COMMIT record has been received, the snapshot should still see the transaction as in-progress.

Reply:
I confirm that this test failed without this change.
So what's happening: 
- replica receives INSERT WAL record in pg_class for created table 't'. Transaction is not committed yet
- we run query `select * from pg_class` at replica which set XMIN_INVALID hint bit for this tuple
- replica receives COMMIT record. But tuple already has XMIN_INVALID hint bit and so is not visible by MVCC